src/test/java/com/squareup/javapoet/TypeNameTest.java

```diff
@@ -99,4 +100,14 @@ private void assertEqualsHashCodeAndToString(TypeName a, TypeName b) {
     assertThat(a.hashCode()).isEqualTo(b.hashCode());
   }
 
+  @Test public void binary() {
+    assertEquals("void", TypeName.VOID.binary());
+    assertEquals("int", TypeName.INT.binary());
+    assertEquals("java.lang.Object", TypeName.OBJECT.binary());
+    assertEquals("java.lang.Thread$State", TypeName.get(Thread.State.class).binary());
+    assertEquals("[java.util.Map$Entry;", TypeName.get(Map.Entry[].class).binary());
```

Comment:
The Javadoc is over here: https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getName()